### ISSUE TITLE
core.sys.posix.time's timer_t should be void* on linux, not int

### ISSUE THREAD

**Issue by Blub** [ldc-developers/ldc#1983]
Noticed on ArchLinux x86_64. Using `int` for `timer_t` is problematic since `timer_create` takes a pointer to it which will be filled out and `int.sizeof=4` on x86_64 whereas `timer_t` being a pointer itself should be 8 bytes.
https://github.com/ldc-developers/druntime/blob/ldc/src/core/sys/posix/time.d#L197
I don't know about the other platforms, though.

**Comment by klickverbot** [comment]
Hi @Blub,

Thanks for the report! This has already been fixed in upstream as dlang/druntime@2c910837f, and will trickle down into LDC once 2.072 is merged into master.

**Comment by Blub** [comment]
Ah thanks, didn't see that one. (New to D and also missed the 'forked from' line in ldc-developers/druntime otherwise I would have looked through dlang/druntime more closely.)
Sorry for the noise.

**Comment by klickverbot** [comment]
No worries - it can be confusing to keep track of the different versions, and LDC will hopefully catch up to DMD again soon.
